### PR TITLE
Add support for Popup.LightDismissOverlayMode

### PR DESCRIPTION
**Pull request by davidjohnoliver**
GitHub Issue (If applicable): #2081

This allows to configure whether a darkened background should be shown behind a Popup. 

This provides a forward path for the previous workaround of adding a background to the Popup.Child content, which breaks light-dismiss after recent pointer-handling updates.

## PR Type

- Feature
What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Popup/Popup.cs [hunk 1]
```diff
@@ -21,6 +22,40 @@ public partial class Popup : PopupBase
...
+		/// Returns true if the popup should show the light-dismiss overlay with its current configuration, false if not
+		/// </summary>
+		private bool ShouldShowLightDismissOverlay
+		{
+			get
+			{
+				switch (LightDismissOverlayMode)
+				{
+					case LightDismissOverlayMode.Auto:
+						return false; //TODO: establish sensible platform defaults
```
Comment (agneszitte-nventive) [inline src/Uno.UI/UI/Xaml/Controls/Popup/Popup.cs]: Every time a "TODO" is added it's not related to an Issue or a task?

File: src/Uno.UI/UI/Xaml/Style/Generic/SystemResources.xaml [hunk 2]
```diff
@@ -1002,6 +1002,19 @@
 			<StaticResource x:Key="AccentButtonBorderBrushPointerOver" ResourceKey="SystemControlHighlightBaseMediumLowBrush" />
 			<StaticResource x:Key="AccentButtonBorderBrushPressed" ResourceKey="SystemControlHighlightTransparentBrush" />
 			<StaticResource x:Key="AccentButtonBorderBrushDisabled" ResourceKey="SystemControlDisabledTransparentBrush" />
+			
+			<!-- Light Dismiss Overlay Resources -->
+			<!--<StaticResource x:Key="AppBarLightDismissOverlayBackground" ResourceKey="SystemControlPageBackgroundMediumAltMediumBrush" />-->
```
Comment (agneszitte-nventive) [inline src/Uno.UI/UI/Xaml/Style/Generic/SystemResources.xaml]: Why some of them are commented?

**Comment by MatFillion** [comment]
We should add a note in Timepicker.md and DatePicker.md for the LightDismiss properties.

File: src/Uno.UI/UI/Xaml/Controls/Popup/Popup.cs (hunk 1 above)
Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Controls/Popup/Popup.cs]: This was a TODO for the current PR, it's been updated

File: src/Uno.UI/UI/Xaml/Style/Generic/SystemResources.xaml (hunk 2 above)
Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Style/Generic/SystemResources.xaml]: Those ones are already present in the separate AppBar.xaml and CommandBar.xaml files. I left them there for easier tracking of upstream changes.

**Comment by davidjohnoliver** [comment]
> If you can, try to avoid deleting generated files (LightDismissOverlayMode.cs)

Fixed

**Pull request merged**